site/content/entry/citra-progress-report-2017-september.md

```diff
@@ -0,0 +1,360 @@
+number. Although it may seem like a very esoteric function, it's actually extremely
+useful in fields like linear algebra, databases, and what the 3DS uses it for:
+texture mapping.
+
+Computers have an intermediate chunk of memory between RAM and the CPU called a
+cache. Caches are seperated into lines, each of which can hold one data item. GPUs
+also have a cache, also seperated into lines. Because they are seperated like this,
+if a texture is loaded into the cache, it would have to span multiple cache lines,
+or even not fit into the cache completely, thus making transformations on it slow,
+as it would have to load and store pieces of it from RAM multiple times.
+
+To avoid this, GPUs can Morton encode textures so that two-dimensional manipulations
+are more likely to only need data already in the cache. Textures that have been
+Morton coded are usually referred to as swizzled or twiddled textures.
+
+{{< figure src="/images/entry/citra-progress-report-2017-september/morton-koopa.png" 
+    title="Not this Morton!" >}}
+<!-- TODO: Right-align image and have text flow down the left. -->
+
+In the function that Morton is implemented, there was a lookup table on Morton
+codes in the comments, and [huwpascoe](https://github.com/huwpascoe) thought it'd
+be best if we just use the lookup table directly. It worked just as well as before,
+but it only needed to perform less than a third of the math. Because this function
+is called so often during emulation (a rough estimate from them is about "millions
+of times a second"), this change although small, made very big changes in CPU performance.
+
+## [Add draw for immediate and batch modes](https://github.com/citra-emu/citra/pull/2921) by [jroweboy](https://github.com/jroweboy)
+
+The 3DS' GPU has two main modes for drawing to the screen, immediate and batch
+mode. In the former, the GPU takes and immediately draws every vertex as it is
+handed to it. In the latter, the GPU accepts vertices given to it, but doesn't
+actually bother drawing them until absolutely necessary, saving a bit of time
+from not having to go through the drawing procedure for every individual vertex.
+Although most games don't use immediate mode at all due to it being extremely
+slow, a handful do use it for a handful of visual effects, like Super Mario Bros. 2.
+
+About a year ago when the GPU code on Citra was rewritten, a handful of calls to
+the drawing routine were removed, as it was believed they were unnecessary. Turns
+out, one of the calls was actually needed for some effects in games, as it handled
+immediate mode drawing. This wasn't noticed for a very long time, as most games
+appeared to carry on with no side-effects at all from the rewrite, but was eventually
+found after some research courtesy of [ds84182](https://github.com/ds84182).
+
+## [Interpolate on a frame-by-frame basis](https://github.com/citra-emu/citra/pull/2858) by [MerryMage](https://github.com/MerryMage)
```

Comment:
IMO `Interpolate audio samples on a frame-by-frame basis` is a more fitting name